File: chapter2/src/jisho.py
import click

@click.command()
@click.argument('text')
def search(text):
    click.echo(f'Searching for {text}')
    dicfile = "ejdict-hand-utf8.txt"
    
    try:
        with open(dicfile, "r", encoding="utf-8") as f:
            found = False
            for line in f:
                if text.lower() in line.lower():
                    click.echo(line.strip())
                    found = True
            if not found:
                click.echo("No results found.")
    except FileNotFoundError:
        click.echo(f"Dictionary file '{dicfile}' not found.")

if __name__ == "__main__":
    search()

File: chapter2/src/jisho.rs
use std::env;
use std::fs::File;
use std::io::{BufRead, BufReader};

pub fn run() {
    let args: Vec<String> = env::args().collect();
    let text: &str = &args[1].trim();
    println!("Searching for {}", text);
    let dicfile = match File::open("ejdict-hand-utf8.txt") {
        Ok(file) => file,
        Err(e) => {
            eprintln!("Error: {}", e);
            return;
        }
    };
    let reader = BufReader::new(dicfile);
    let mut found = false;
    for line in reader.lines() {
        let line = line.unwrap();
        if line.contains(text) {
            println!("{}", line);
            found = true;
        }
    }
    if found == false {
        println!("No match found");
    }
}
